lib/u3d/utils.rb

```diff
@@ -151,6 +151,23 @@ def ensure_dir(dir)
         FileUtils.mkpath(dir) unless File.directory?(dir)
       end
 
+      def get_write_access(dir)
+        if U3dCore::Helper.operating_system == :win
+          yield
+        else
+          owner, access = U3dCore::CommandExecutor.execute(command: "stat -f \"%Su,%A\" #{dir}", admin: false).strip.split(',')
```

Comment:
not sure this works on  Linux. Maybe you need:
<img width="648" alt="image" src="https://user-images.githubusercontent.com/24282/68880411-7f4cdd80-070b-11ea-9b30-bcfd4e91ef41.png">

Also if you  are on a Mac and have the GNU core utils before the Mac one you could have problems:

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/24282/68880542-bb803e00-070b-11ea-8fcb-23642baa8df0.png">